### PR TITLE
Added ES/Solr dependencies to javadoc maven plugin

### PR DESCRIPTION
**Pull request by barthanssens**
This PR addresses GitHub issue: eclipse/rdf4j#1148 .

Briefly describe the changes proposed in this PR:

* Set source version to 8
* Added ES and Solr dependencies to javadoc maven plugin in parent pom.xml
* Moved ES and Solr versions to parent pom.xml

Even though 'maven clean install -DskipTests' works, running 'maven clean install -DskipTests -Passembly' fails in rdf4j-storage/storage (in attach-javadocs), with 90+ errors about missing packages and symbols.

Actually this is already an issue in JDK8, and can be seen when debugging/logging, but javadoc 8 just continues after the error, so the maven goal succeeds.
But with JDK9 (and 11), it's a fatal error (even with -Xdoclint=none)

Not the prettiest patch, but I couldn't get it to work otherwise (e.g. moving this to solr/pom.xml and elasticsearch/pom.xml didn't seem to work)

Also note that the javadoc maven plugin 2.8.1 uses misspelled "additionnalDependencies" with extra "n", instead of "additionalDependencies" (fixed in newer versions)

CQs:
- [x] CQ 20696 ElasticSearch (server) 6.5.0
- [x] CQ 20697 ElasticSearch X-Content 6.5.0
- [x] CQ 20698  Solr Core 7.5.0

### CODE REVIEW

File: elasticsearch/pom.xml [hunk 1]
```diff
@@ -43,4 +39,32 @@
...
+							<version>${elasticsearch.version}</version>
+						</additionnalDependency>
+						<additionnalDependency>
+							<groupId>org.elasticsearch</groupId>
+							<artifactId>elasticsearch-x-content</artifactId>
+							<version>${elasticsearch.version}</version>
+						</additionnalDependency>
+						<additionnalDependency>
+							<groupId>org.elasticsearch</groupId>
+							<artifactId>elasticsearch</artifactId>
```
Comment (jeenbroekstra) [inline elasticsearch/pom.xml]: You said 'elasticsearch' twice :)

**Comment by hmottestad** [comment]
Could we bump the javadoc version at the same time to handle that typo?

**Comment by barthanssens** [comment]
Indeed, good catch.
We don't have a CQ for that, the ES-X-Content is a dependency of elasticsearch (core), which in turn is a dependency of the optional elasticsearch transport... So this probably requires a "works-with" CQ

(IIRC, it's not the doclinter but the compiler that causes the issue, since the javadoc plugin runs the compile a second time. Might be mitigated by moving from 2.8.1 to the latest javadoc version 3.1.1 that does include more options for not running compile again, but that might require more changes)

File: elasticsearch/pom.xml (hunk 1 above)
Comment (barthanssens) [reply elasticsearch/pom.xml]: Whoops :-)

**Comment by barthanssens** [comment]
> Could we bump the javadoc version at the same time to handle that typo?

We can upgrade to 2.10.4 (no CQ required, already covered in the range 2.8.1-3.1.0 "works-with" dependency part of CQ20318)

I'd suggest to do that in a separate branch, since the versions of the other projects should be kept in sync (and I suppose there is no impact on the jenkins builds)

3.1.x is also an option, but probably requires a bit more changes and testing...

**Comment by jeenbroekstra** [comment]
Given that this has pending CQs it will not be included in the 3.0.0 release.

**Comment by jeenbroekstra** [comment]
I have tagged the 3.0 release, so we can merge this to develop now. @barthanssens are you happy for me to do so? I want to get this merged to (eventually) master (as part of the develop sync) so that it can be included in the mono-repo migration.

**Comment by barthanssens** [comment]
> Given that this has pending CQs it will not be included in the 3.0.0 release.

According to IPZilla, the last CQ has just been approved, so one step close to full JDK11 support ;-)

**Pull request merged**